tools/libSwiftScan/libSwiftScan.cpp

```diff
@@ -684,6 +685,24 @@ swiftscan_cas_t swiftscan_cas_create(const char *path) {
   return wrap(tool);
 }
 
+swiftscan_cas_t swiftscan_cas_create_from_plugin(
+    const char *plugin_path, const char *ondisk_path, int argc,
+    const char **keys, const char **values) {
+  clang::CASOptions CASOpts;
+  CASOpts.PluginPath = plugin_path;
+  if (ondisk_path)
+    CASOpts.CASPath = ondisk_path;
+  for (int i = 0; i < argc; ++i) {
+    CASOpts.PluginOptions.emplace_back(keys[i], values[i]);
+  }
+  CachingTool *tool = new CachingTool(CASOpts);
```

Comment:
I'd suggest to use `std::unique_ptr` instead of manual new/delete and call `.release()` on it before returning it.

Reply:
I keep this as shared_ptr so that cas_option_t and cas_t can be separately destroyed.

Reply:
I realize I was mis-understanding the comment. I keep new/delete as this is the convention in the file.